logstash-core/lib/logstash/api/rack_app.rb

```diff
@@ -8,9 +8,84 @@
 module LogStash
   module Api
     module RackApp
-      def self.app
+      # Cabin is not compatible with CommonLogger, and this gives us more control anyway
+      class ApiLogger
```

Comment:
I think the performance impact here is negligible. You would need to get to thousands of TPS before this rack middleware had a measurable impact in my estimation.

I agree this still needs tests, that's why its marked as a WIP.